### PR TITLE
Remove references to external teams (and other light cleanup)

### PR DESCRIPTION
**Pull request by batemapf**
In an effort to get VFS team content cleaned up before it starts getting pulled into `va.gov-team`:

- Removes outdated External vs. Internal vs. DSVA classification
- Updates Slack channel references
- First pass at removing some minor extraneous content

### CODE REVIEW

**Comment by batemapf** [comment]
@rroueche not totally sure where / how to route this, so tagging you :-)

File: DeveloperDocs/vets-website/architecture.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ If you're a developer, most of your time will likely be spent working on a React
...
 
-You will also find that our visual components and site-wide styles live in an external [design system](https://department-of-veterans-affairs.github.io/design-system/). This design system is based on the [US Web Design System](https://designsystem.digital.gov/), with some additional modifications by our team. As part of that design system, we provide a set of React components covering common patterns, which is [published to npm](https://www.npmjs.com/package/@department-of-veterans-affairs/jean-pants). You'll likely use these components to make your apps accessible and consistent with the rest of Vets.gov.
+You will also find that our visual components and site-wide styles live in their own [design system](https://design.va.gov). This design system is based on the [US Web Design System](https://designsystem.digital.gov/), with some additional modifications by our team. As part of that design system, we provide a set of React components covering common patterns, which is [published to npm](https://www.npmjs.com/package/@department-of-veterans-affairs/jean-pants). You'll likely use these components to make your apps accessible and consistent with the rest of Vets.gov.
```
Comment (annaswims) [inline DeveloperDocs/vets-website/architecture.md]: `jean-pants` is now `formation` https://www.npmjs.com/package/@department-of-veterans-affairs/formation

Comment (batemapf) [reply DeveloperDocs/vets-website/architecture.md]: thanks, updated!

**Comment by batemapf** [comment]
@smoonpeyes it looks like that doc hasn't been edited since oct 2018, so probably it is out of date and needs to be addressed. not going to attempt to do that in this PR, though :-)

**Pull request merged**